x-pack/plugins/uptime/public/hooks/use_composite_image.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { composeScreenshotRef } from '../lib/helper/compose_screenshot_images';
+import { ScreenshotRefImageData } from '../../common/runtime_types/ping/synthetics';
+
+/**
+ * Checks if two refs are the same. If the ref is unchanged, there's no need
+ * to run the expensive draw procedure.
+ */
+function isNewRef(a?: ScreenshotRefImageData, b?: ScreenshotRefImageData): boolean {
+  if (typeof a === 'undefined' || typeof b === 'undefined') return false;
+  const stepA = a.ref.screenshotRef.synthetics.step;
+  const stepB = b.ref.screenshotRef.synthetics.step;
+  return stepA.index !== stepB.index || stepA.name !== stepB.name;
+}
+
+/**
+ * Assembles the data for a composite image and returns the composite to a callback.
+ * @param imgRef the data and dimensions for the composite image.
+ * @param callback sends the composited image to this callback.
+ * @param url this is the composited image value, if it is truthy the function will skip the compositing process
+ */
+export const useCompositeImage = (
+  imgRef: ScreenshotRefImageData,
+  callback: React.Dispatch<string | undefined>,
```

Comment:
```suggestion
  onComposeImage: React.Dispatch<string | undefined>,
```

Reply:
🥇

Reply:
I actually am going to call it `onComposeImageSuccess` because it only gets called in the event of a successful run; if there's a failure it won't call the callback, since the script will fail with a rejected promise.